### PR TITLE
New project

### LINKED ISSUE #66: Replace all placeholders with actual values
Loading of a project now works, all the placeholders can now be removed.

### PR DESCRIPTION
**Pull request by mennoo1996**
Fixes #66 
You can now create a new project via File -> New
At program start, there aren't some random amount of timelines anymore, but you have to create or load a project first
On project create, you can add some camera types, cameras and timelines with the correct names and settings.
Next step is to be able to change these settings after create, but that's for a different pull request

### CODE REVIEW

File: src/main/java/control/ControllerManager.java [hunk 1]
```diff
@@ -65,6 +73,10 @@ public ControllerManager(RootPane rootPane, TimelineController timelineControlle
         this.detailViewController = detailViewController;
         this.toolViewController = toolViewController;
     }
+    
+
```
Comment (bart-de-jonge) [inline src/main/java/control/ControllerManager.java]: What is all this whitespace? Please remove

File: src/main/java/control/ControllerManager.java [hunk 2]
```diff
@@ -73,7 +85,10 @@ private void initializeControllers() {
         timelineControl = new TimelineController(this);
         detailViewController = new DetailViewController(this);
         toolViewController = new ToolViewController(this);
+        fileMenuController = new FileMenuController(this);
     }
+    
+
```
Comment (bart-de-jonge) [inline src/main/java/control/ControllerManager.java]: Unnecessary whitespace

File: src/main/java/control/FileMenuController.java [hunk 3]
```diff
@@ -0,0 +1,150 @@
...
+    
+    /**
+     * Save the current project state to file.
+     * A file chooser window will be opened to select a file
+     */
+    public void save() {
+        log.info("Saving Project to file");
+        FileChooser fileChooser = new FileChooser();
+        fileChooser.setTitle("Save");
+        ExtensionFilter extFilter = new ExtensionFilter("txt files", "*.txt");
```
Comment (bart-de-jonge) [inline src/main/java/control/FileMenuController.java]: .txt extension is not very professional, could you change it to something more suitable (maybe define an own one)

File: src/main/java/control/FileMenuController.java [hunk 4]
```diff
@@ -0,0 +1,150 @@
...
+        fileChooser.getExtensionFilters().add(extFilter);
+        File file = fileChooser.showSaveDialog(controllerManager.getRootPane().getPrimaryStage());
+        if (file != null) {
```
Comment (bartdejonge1996) [inline src/main/java/control/FileMenuController.java]: what happens if the file is actually null? At least log an error, maybe some feedback to the user

File: src/main/java/control/FileMenuController.java [hunk 5]
```diff
@@ -0,0 +1,150 @@
...
+     * A file chooser window will be opened to select the file.
+     */
+    public void load() {
+        log.info("Loading Project from file");
+        FileChooser fileChooser = new FileChooser();
+        fileChooser.setTitle("Load");
+        ExtensionFilter extFilter = new ExtensionFilter("txt files", "*.txt");
+        fileChooser.getExtensionFilters().add(extFilter);
+        File file = fileChooser.showOpenDialog(controllerManager.getRootPane().getPrimaryStage());
+        if (file != null) {
```
Comment (bart-de-jonge) [inline src/main/java/control/FileMenuController.java:77]: Same story, what if it is null

File: src/main/java/control/FileMenuController.java [hunk 6]
```diff
@@ -0,0 +1,150 @@
...
+    
+    /**
+     * Load a project from file.
+     * A file chooser window will be opened to select the file.
+     */
+    public void load() {
+        log.info("Loading Project from file");
+        FileChooser fileChooser = new FileChooser();
+        fileChooser.setTitle("Load");
+        ExtensionFilter extFilter = new ExtensionFilter("txt files", "*.txt");
```
Comment (bart-de-jonge) [inline src/main/java/control/FileMenuController.java]: new extension?

File: src/main/java/control/FileMenuController.java [hunk 7]
```diff
@@ -0,0 +1,150 @@
...
+            ScriptingProject temp  = ScriptingProject.read(file);
+            if (temp == null) {
+                Alert alert = new Alert(AlertType.ERROR); 
+                alert.setTitle("Load Failed");
+                alert.setContentText("The format in the selected file was not recognized");
+                alert.showAndWait();
+            } else {
+                controllerManager.setScriptingProject(temp);
+                controllerManager.getRootPane()
+                                 .reInitRootCenterArea(new RootCenterArea(
```
Comment (bart-de-jonge) [inline src/main/java/control/FileMenuController.java]: this indentation is ridiculous, please use only 1 tab at a time. You could also initialise the rootcenterArea on a seperate line for readability

File: src/main/java/control/FileMenuController.java [hunk 8]
```diff
@@ -0,0 +1,150 @@
...
+        log.info("Loading Project from file");
+        FileChooser fileChooser = new FileChooser();
+        fileChooser.setTitle("Load");
+        ExtensionFilter extFilter = new ExtensionFilter("txt files", "*.txt");
+        fileChooser.getExtensionFilters().add(extFilter);
+        File file = fileChooser.showOpenDialog(controllerManager.getRootPane().getPrimaryStage());
+        if (file != null) {
+            ScriptingProject temp  = ScriptingProject.read(file);
+            if (temp == null) {
+                Alert alert = new Alert(AlertType.ERROR);
```
Comment (bart-de-jonge) [inline src/main/java/control/FileMenuController.java:80]: You could use this when the files are null

File: src/main/java/control/FileMenuController.java [hunk 9]
```diff
@@ -0,0 +1,150 @@
...
+     * @param project the project that was loaded
+     */
+    public void addLoadedBlocks(ScriptingProject project) {
+        log.info("Adding loaded blocks");
+        for (int i = 0; i < project.getCameraTimelines().size();i++) {
+            CameraTimeline timeline = project.getCameraTimelines().get(i);
+            int amountShots = timeline.getShots().size();
+            for (int j = 0; j < amountShots;j++) {
+                CameraShot shot = timeline.getShots().get(j);
+                addCameraShotForLoad(i, shot);
```
Comment (bart-de-jonge) [inline src/main/java/control/FileMenuController.java]: Maybe inline the shot variable?

File: src/main/java/control/FileMenuController.java [hunk 10]
```diff
@@ -0,0 +1,150 @@
...
+        }
+    }
+    
+    /**
+     * Add a camera shot that is loaded from file.
+     * @param cameraIndex the index of the camera timeline to use
+     * @param shot the shot to add
+     */
+    private void addCameraShotForLoad(int cameraIndex, CameraShot shot) {
+        CameraShotBlock shotBlock = new CameraShotBlock(shot, cameraIndex,
```
Comment (bart-de-jonge) [inline src/main/java/control/FileMenuController.java]: indentation

File: src/main/java/control/FileMenuController.java [hunk 11]
```diff
@@ -0,0 +1,150 @@
...
+    private void addCameraShotForLoad(int cameraIndex, CameraShot shot) {
+        CameraShotBlock shotBlock = new CameraShotBlock(shot, cameraIndex, 
+                                                       controllerManager.getRootPane()
+                                                                        .getRootCenterArea(),
+                                                       controllerManager.getTimelineControl()
+                                                                        ::shotChangedHandler);
+        controllerManager.setActiveShotBlock(shotBlock);
+        controllerManager.getTimelineControl().getCameraShotBlocks().add(shotBlock);
+    }
+
```
Comment (bart-de-jonge) [inline src/main/java/control/FileMenuController.java:136]: missing javadoc

File: src/main/java/control/FileMenuController.java [hunk 12]
```diff
@@ -0,0 +1,150 @@
...
+                                                                        ::shotChangedHandler);
+        controllerManager.setActiveShotBlock(shotBlock);
+        controllerManager.getTimelineControl().getCameraShotBlocks().add(shotBlock);
+    }
+    
+    public void newProject() {
+        new NewProjectModalView(controllerManager.getRootPane(), this::createProject);
+    }
+    
+    public void newProject(MouseEvent event) {
```
Comment (bart-de-jonge) [inline src/main/java/control/FileMenuController.java:153]: missing javadoc

File: src/main/java/control/FileMenuController.java [hunk 13]
```diff
@@ -0,0 +1,150 @@
...
+        project.setCameraTimelines(event.getTimelines());
+        project.getDirectorTimeline().setProject(project);
+        for (CameraTimeline timeline : project.getCameraTimelines()) {
+            timeline.setProject(project);
+        }
+        
+        controllerManager.setScriptingProject(project);
+        RootCenterArea area = new RootCenterArea(controllerManager.getRootPane(),
+                                                 event.getTimelines().size(),
+                                                 false);
```
Comment (bart-de-jonge) [inline src/main/java/control/FileMenuController.java:253]: This should be possible to add on the previoius line

File: src/main/java/control/TimelineController.java [hunk 14]
```diff
@@ -80,7 +69,10 @@ public void addCameraShot(int cameraIndex, String name, String description,
         log.info("Adding CameraShot to Timeline");
 
         CameraShot newShot = new CameraShot(name,description, startCount, endCount);
-        this.project.getCameraTimelines().get(cameraIndex).addShot(newShot);
+        this.controllerManager.getScriptingProject()
+                              .getCameraTimelines()
+                              .get(cameraIndex)
```
Comment (bart-de-jonge) [inline src/main/java/control/TimelineController.java:74]: you could merge some of these calls on the same line right?

File: src/main/java/control/TimelineController.java [hunk 15]
```diff
@@ -132,12 +114,14 @@ public void removeCameraShot(CameraShotBlock cameraShotBlock) {
      */
     public void shotChangedHandler(CameraShotBlockUpdatedEvent event) {
         log.info("Shot moved to new TimeLine");
+        log.error("Handling changed shot");
```
Comment (bart-de-jonge) [inline src/main/java/control/TimelineController.java]: why do you log an error here?

File: src/main/java/control/TimelineController.java [hunk 16]
```diff
@@ -178,7 +163,9 @@ private void checkCollisions(int timelineNumber, CameraShotBlock cameraShotBlock
      */
     private void checkCollisions(int timelineNumber, int oldTimelineNumber,
                                  CameraShotBlock cameraShotBlock) {
-        CameraTimeline timeline = project.getCameraTimelines().get(timelineNumber);
+        CameraTimeline timeline = controllerManager.getScriptingProject()
+                                                    .getCameraTimelines()
```
Comment (bart-de-jonge) [inline src/main/java/control/TimelineController.java]: merge the last two calls on one line

File: src/main/java/control/TimelineController.java [hunk 17]
```diff
@@ -234,75 +219,11 @@ private void removeCollisionFromCameraShotBlock(CameraShotBlock shotBlock) {
...
-     */
-    private void initializeCameraTimelines() {
-        log.info("Initializing camera Timelines");
-        for (int i = 0; i < numTimelines; i++) {
-            Camera defCam = new Camera("IP Cam " + i, "", defType);
-            CameraTimeline timelineN = new CameraTimeline(defCam, "", project);
-            project.addCameraTimeline(timelineN);
-        }
-    }
+
```
Comment (bart-de-jonge) [inline src/main/java/control/TimelineController.java]: loads of whitespace here

File: src/main/java/gui/modal/AddCameraModalView.java [hunk 18]
```diff
@@ -0,0 +1,126 @@
...
+     */
+    private void initializeView() {
+        this.viewPane = new VBox(20);
+        this.viewPane.getChildren().add(new Text("Create a camera"));
+        
+        initFields();
+        initTypeList();
+        
+        addCameraButton = new Button("Add Camera");
+        addCameraButton.setOnMouseClicked(this::addCamera);
```
Comment (bart-de-jonge) [inline src/main/java/gui/modal/AddCameraModalView.java]: The idea of mvc is to add the handlers directly in the controllers to prevent overhead for unnecesary events. On the click of this button you make an event, fire it and collect it at the controller. If you just add the handler directly in the controller this overhed will be removed

File: src/main/java/gui/events/AddCameraEvent.java [hunk 19]
```diff
@@ -0,0 +1,35 @@
+package gui.events;
+
+import data.CameraType;
+import javafx.event.Event;
+import javafx.event.EventType;
+import lombok.Getter;
+
+public class AddCameraEvent extends Event {
```
Comment (bart-de-jonge) [inline src/main/java/gui/events/AddCameraEvent.java]: unnecessary event, see comments below on controllers and handlers

File: src/main/java/gui/events/AddCameraTypeEvent.java [hunk 20]
```diff
@@ -0,0 +1,34 @@
+package gui.events;
+
+import javafx.event.Event;
+import javafx.event.EventType;
+import lombok.Getter;
+
+public class AddCameraTypeEvent extends Event {
```
Comment (bart-de-jonge) [inline src/main/java/gui/events/AddCameraTypeEvent.java]: unnecessary event, see comments below on controllers and handlers

File: src/main/java/gui/events/AddTimelineEvent.java [hunk 21]
```diff
@@ -0,0 +1,29 @@
+package gui.events;
+
+import data.Camera;
+import javafx.event.Event;
+import javafx.event.EventType;
+import lombok.Getter;
+
+public class AddTimelineEvent extends Event {
```
Comment (bart-de-jonge) [inline src/main/java/gui/events/AddTimelineEvent.java]: unnecessary event, see comments below on controllers and handlers

File: src/main/java/gui/events/NewProjectCreationEvent.java [hunk 22]
```diff
@@ -0,0 +1,52 @@
...
+
+import java.util.ArrayList;
+
+import data.Camera;
+import data.CameraTimeline;
+import javafx.event.Event;
+import javafx.event.EventType;
+import lombok.Getter;
+
+public class NewProjectCreationEvent extends Event {
```
Comment (bart-de-jonge) [inline src/main/java/gui/events/NewProjectCreationEvent.java]: unnecessary event, see comments below on controllers and handlers

File: src/main/java/gui/headerarea/DoubleTextField.java [hunk 23]
```diff
@@ -0,0 +1,30 @@
...
+    }
+
+    @Override
+    public void replaceSelection(String text) {
+        if (validate(this.getText())) {
+            super.replaceSelection(text);
+        }
+    }
+
+    private boolean validate(String text) {
```
Comment (bart-de-jonge) [inline src/main/java/gui/headerarea/DoubleTextField.java:39]: missing javadoc

File: src/main/java/gui/headerarea/DoubleTextField.java [hunk 24]
```diff
@@ -0,0 +1,30 @@
...
+        String newText = this.getText().substring(0, start) + text
+                + this.getText().substring(end, this.getText().length());
+
+        if (validate(newText)) {
+            super.replaceText(start, end, text);
+        }
+    }
+
+    @Override
+    public void replaceSelection(String text) {
```
Comment (bart-de-jonge) [inline src/main/java/gui/headerarea/DoubleTextField.java:28]: missing javadoc

File: src/main/java/gui/headerarea/DoubleTextField.java [hunk 25]
```diff
@@ -0,0 +1,30 @@
...
+
+import javafx.scene.control.TextField;
+
+/**
+* Created by Bart.
+*/
+public class DoubleTextField extends TextField {
+
+    @Override
+    public void replaceText(int start, int end, String text) {
```
Comment (bart-de-jonge) [inline src/main/java/gui/headerarea/DoubleTextField.java:15]: missing javadoc

File: src/main/java/gui/modal/AddCameraTypeModalView.java [hunk 26]
```diff
@@ -0,0 +1,107 @@
...
+     * Initialize the view of this modal.
+     */
+    private void initializeView() {
+        this.viewPane = new VBox(20);
+        this.viewPane.getChildren().add(new Text("Create a camera type"));
+        
+        initFields();
+        
+        addCameraTypeButton = new Button("Add Camera Type");
+        addCameraTypeButton.setOnMouseClicked(this::addCameraType);
```
Comment (bart-de-jonge) [inline src/main/java/gui/modal/AddCameraTypeModalView.java]: please add the handler in the controller and remove this event

File: src/main/java/gui/modal/AddTimelineModalView.java [hunk 27]
```diff
@@ -0,0 +1,117 @@
...
+     */
+    private void initializeView() {
+        this.viewPane = new VBox(20);
+        this.viewPane.getChildren().add(new Text("Create a camera timeline"));
+        
+        initFields();
+        initCameraList();
+        
+        addTimelineButton = new Button("Add Timeline");
+        addTimelineButton.setOnMouseClicked(this::addTimeline);
```
Comment (bart-de-jonge) [inline src/main/java/gui/modal/AddTimelineModalView.java]: please add the handler in the controller and remove this event

File: src/main/java/gui/modal/NewProjectModalView.java [hunk 28]
```diff
@@ -0,0 +1,233 @@
...
+        this.viewPane = new VBox(20);
+        this.viewPane.getChildren().add(new Text("Create a new project"));
+        
+        initFields();
+        initAddCameraType();
+        initAddCamera();
+        initAddTimeline();
+        
+        creationButton = new Button("Create new project");
+        creationButton.setOnMouseClicked(this::createProject);
```
Comment (bart-de-jonge) [inline src/main/java/gui/modal/NewProjectModalView.java]: please add the handler in the controller and remove this event

File: src/main/java/gui/modal/NewProjectModalView.java [hunk 29]
```diff
@@ -0,0 +1,233 @@
...
+        super.setModalView(this.viewPane);
+        super.displayModal();
+    }
+
+    /**
+     * Initialize the section to add timelines.
+     */
+    private void initAddTimeline() {
+        addTimelineButton = new Button("Add Camera Timeline");
+        addTimelineButton.setOnMouseClicked(this::addTimeline);
```
Comment (bart-de-jonge) [inline src/main/java/gui/modal/NewProjectModalView.java]: please add the handler in the controller and remove this event

File: src/main/java/gui/modal/NewProjectModalView.java [hunk 30]
```diff
@@ -0,0 +1,233 @@
...
+        
+        timelineList = new ListView<Label>();
+        timelineList.setMaxHeight(100);
+        
+
```
Comment (bart-de-jonge) [inline src/main/java/gui/modal/NewProjectModalView.java]: whitespace

File: src/main/java/gui/modal/NewProjectModalView.java [hunk 31]
```diff
@@ -0,0 +1,233 @@
...
+    private void initAddCamera() {
+        addCameraButton = new Button("Add Camera");
+        addCameraButton.setOnMouseClicked(this::addCamera);
+        
+        cameraList = new ListView<Label>();
+        cameraList.setMaxHeight(100);
+        
+        this.viewPane.getChildren().addAll(addCameraButton, cameraList);
+
+
```
Comment (bart-de-jonge) [inline src/main/java/gui/modal/NewProjectModalView.java]: whitespace

File: src/main/java/gui/modal/NewProjectModalView.java [hunk 32]
```diff
@@ -0,0 +1,233 @@
...
+        
+        this.viewPane.getChildren().addAll(addCameraTypeButton, cameraTypeList);
+    }
+
+    /**
+     * Initialize the section to add cameras.
+     */
+    private void initAddCamera() {
+        addCameraButton = new Button("Add Camera");
+        addCameraButton.setOnMouseClicked(this::addCamera);
```
Comment (bart-de-jonge) [inline src/main/java/gui/modal/NewProjectModalView.java]: please add the handler in the controller and remove this event

File: src/main/java/gui/modal/NewProjectModalView.java [hunk 33]
```diff
@@ -0,0 +1,233 @@
...
+        this.viewPane.getChildren().addAll(addTimelineButton, timelineList);
+        
+    }
+    
+    /**
+     * Initialize the section to add camera types.
+     */
+    private void initAddCameraType() {
+        addCameraTypeButton = new Button("Add Camera Type");
+        addCameraTypeButton.setOnMouseClicked(this::addCameraType);
```
Comment (bart-de-jonge) [inline src/main/java/gui/modal/NewProjectModalView.java]: please add the handler in the controller and remove this event

File: src/main/java/gui/modal/NewProjectModalView.java [hunk 34]
```diff
@@ -0,0 +1,233 @@
...
+     */
+    private void handleAddTimeline(AddTimelineEvent event) {
+        CameraTimeline timeline = new CameraTimeline(event.getCamera(),
+                                                     event.getDescription(),
+                                                     null);
+        timelines.add(timeline);
+        timelineList.getItems().add(new Label(timeline.getDescription()));
+    }
+    
+    private void addCameraType(MouseEvent event) {
```
Comment (bart-de-jonge) [inline src/main/java/gui/modal/NewProjectModalView.java]: missing javadoc, but this will probably be removed

File: src/main/java/gui/root/RootCenterArea.java [hunk 35]
```diff
@@ -62,26 +66,58 @@
...
+    public RootCenterArea(RootPane rootPane, int numberOfTimelines, boolean empty) {
+        if (empty) {
+            this.rootPane = rootPane;
+            this.numberOfTimelines = numberOfTimelines;
+            HBox buttonBox = new HBox();
+            buttonBox.setSpacing(10);
+            Button newButton = new Button("Create new project");
+            Button loadButton = new Button("Load project");
+            buttonBox.getChildren().addAll(newButton, loadButton);
+            newButton.setOnMouseClicked(rootPane.getControllerManager()
```
Comment (bart-de-jonge) [inline src/main/java/gui/root/RootCenterArea.java]: please add the handler in the controller and remove this event

File: src/main/java/gui/root/RootCenterArea.java [hunk 36]
```diff
@@ -62,26 +66,58 @@
...
+                                                .getFileMenuController()::newProject);
+            loadButton.setOnMouseClicked(rootPane.getControllerManager()
```
Comment (bart-de-jonge) [inline src/main/java/gui/root/RootCenterArea.java]: please add the handler in the controller and remove this event

**Comment by bart-de-jonge** [comment]
Posted some comments on a major issue with mvc in the PR, also added some small comments. Other then that these new features look great!

File: src/main/java/control/FileMenuController.java (hunk 3 above)
Comment (mennoo1996) [inline src/main/java/control/FileMenuController.java]: Changed to .scp (SCripting Project)

File: src/main/java/control/FileMenuController.java (hunk 4 above)
Comment (mennoo1996) [inline src/main/java/control/FileMenuController.java:59]: If the file is null, it means the user did not select a file. So I think the user should know that nothing should happen then.

File: src/main/java/control/FileMenuController.java (hunk 6 above)
Comment (mennoo1996) [inline src/main/java/control/FileMenuController.java]: See other comment about extension

File: src/main/java/control/FileMenuController.java (hunk 5 above)
Comment (mennoo1996) [inline src/main/java/control/FileMenuController.java:77]: Again, it means the user did not select a file, so the user should not get any extra error messages because he knows he aborted the file selection himself

File: src/main/java/control/FileMenuController.java (hunk 8 above)
Comment (mennoo1996) [inline src/main/java/control/FileMenuController.java:80]: Not needed, see other comments on this issue

File: src/main/java/control/FileMenuController.java (hunk 7 above)
Comment (mennoo1996) [inline src/main/java/control/FileMenuController.java]: Moved everything a bit to the left

File: src/main/java/control/FileMenuController.java (hunk 9 above)
Comment (mennoo1996) [inline src/main/java/control/FileMenuController.java]: Done

File: src/main/java/control/FileMenuController.java (hunk 13 above)
Comment (mennoo1996) [inline src/main/java/control/FileMenuController.java:253]: I think it's better to put the parameters all below each other in situations like this, rather than 1 on 1 line and 2 on the other line

File: src/main/java/control/FileMenuController.java (hunk 10 above)
Comment (mennoo1996) [inline src/main/java/control/FileMenuController.java]: Fixed

File: src/main/java/control/FileMenuController.java (hunk 11 above)
Comment (mennoo1996) [inline src/main/java/control/FileMenuController.java:136]: Done

File: src/main/java/control/FileMenuController.java (hunk 12 above)
Comment (mennoo1996) [inline src/main/java/control/FileMenuController.java:153]: Done

File: src/main/java/control/TimelineController.java (hunk 14 above)
Comment (mennoo1996) [inline src/main/java/control/TimelineController.java:74]: I think this way it is structured better. It may take a few more lines but the method calls are very nicely aligned

File: src/main/java/control/TimelineController.java (hunk 15 above)
Comment (mennoo1996) [inline src/main/java/control/TimelineController.java]: Forgot to remove, was for debug purposes

File: src/main/java/control/TimelineController.java (hunk 16 above)
Comment (mennoo1996) [inline src/main/java/control/TimelineController.java]: See above

File: src/main/java/control/TimelineController.java (hunk 17 above)
Comment (mennoo1996) [inline src/main/java/control/TimelineController.java]: Removed

File: src/main/java/control/FileMenuController.java (hunk 4 above)
Comment (bart-de-jonge) [inline src/main/java/control/FileMenuController.java:59]: Still, feedback to the user is always nice, maybe popup please select a file. But we can put this in an issue as well

File: src/main/java/gui/headerarea/DoubleTextField.java (hunk 25 above)
Comment (mennoo1996) [inline src/main/java/gui/headerarea/DoubleTextField.java:15]: Fixed

File: src/main/java/gui/headerarea/DoubleTextField.java (hunk 24 above)
Comment (mennoo1996) [inline src/main/java/gui/headerarea/DoubleTextField.java:28]: Fixed

File: src/main/java/gui/headerarea/DoubleTextField.java (hunk 23 above)
Comment (mennoo1996) [inline src/main/java/gui/headerarea/DoubleTextField.java:39]: Fixed

File: src/main/java/gui/modal/NewProjectModalView.java (hunk 30 above)
Comment (mennoo1996) [inline src/main/java/gui/modal/NewProjectModalView.java]: Fixed

File: src/main/java/gui/modal/NewProjectModalView.java (hunk 31 above)
Comment (mennoo1996) [inline src/main/java/gui/modal/NewProjectModalView.java]: Fixed

File: src/test/java/xml/test_files/test-write.xml [hunk 37]
```diff
@@ -1 +1 @@
...
..."?><scripting-project description="A test project" seconds-per-count="2.0"><director-timeline description="A test director timeline"><director-shots number-of-shots="1"><director-shot description="A test director shot" instance="0" name="shot-1"/></director-shots></director-timeline><camera-centerarea><camera-timeline description="Test camera line 1"><camera description="some camera description" movement-margin="4.0" name="Some camera"><camera-type description="b" movement-margin="4.0" name="a"/></camera><camera-shots number-of-shots="1"><camera-shot description="A camera shot" instance="0" name="shot-2"/></camera-shots></camera-timeline><camera-timeline description="Test camera line 2"><camera description="another description" movement-margin="5.0" name="Another camera"><camera-type description="d" movement-margin="5.0" name="c"/></camera><camera-shots number-of-shots="1"><camera-shot description="Another camera shot" instance="1" name="shot-3"/></camera-shots></camera-timeline></camera-centerarea><cameras><camera description="Camera description" movement-margin="4.0" name="Camera"><camera-type description="y" movement-margin="4.0" name="x"/></camera></cameras></scripting-project>
```
Comment (geenen124) [inline src/test/java/xml/test_files/test-write.xml]: Please add test-write.xml to the gitignore

**Comment by bart-de-jonge** [comment]
Great work processign the comments about MVC in our project. Merging the PR
